src/activity_handlers.cpp

```diff
@@ -631,30 +920,131 @@ void activity_handlers::butcher_finish( player_activity *act, player *p )
+        g->m.i_rem( p->pos(), act->index );
+        act->set_to_null();
+        return;
+        break;
+    case F_DRESS:
+        if ( roll_butchery() < 0 ) { // partial failure
+            switch( rng( 1, 3 ) ) {
+                case 1:
+                    p->add_msg_if_player(m_warning, _( "You unskilfully hack the corpse and chop off some excess body parts.  You think this was not a job well done." ) );
+                    break;
+                case 2:
+                    p->add_msg_if_player(m_warning, _( "Your unskilled hands slip and damage the corpse. You still hope it's not a total waste though." ) );
+                    break;
+                case 3:
+                    p->add_msg_if_player(m_warning, _( "You did something wrong and hacked the corpse badly. Maybe it's still recoverable." ) );
+                    break;
+            }
+            corpse_item.set_flag( "FIELD_DRESS_FAILED" );
+
+            g->m.add_splatter( type_gib, p->pos(), rng( corpse->size + 2, ( corpse->size + 1 ) * 2 ) );
+            g->m.add_splatter( type_blood, p->pos(), rng( corpse->size + 2, ( corpse->size + 1 ) * 2 ) );
+            for( int i = 1; i <= corpse->size; i++ ) {
+                g->m.add_splatter_trail( type_gib, p->pos(), random_entry( g->m.points_in_radius( p->pos(), corpse->size + 1 ) ) );
+                g->m.add_splatter_trail( type_blood, p->pos(), random_entry( g->m.points_in_radius( p->pos(), corpse->size + 1 ) ) );
+            }
+
+        } else { // success
+
+            switch( rng( 1, 3 ) ) {
+                case 1:
+                    p->add_msg_if_player( m_good, _("You field dress the %s."), corpse->nname().c_str() );
+                    break;
+                case 2:
+                    p->add_msg_if_player( m_good, _( "You slice the corpse's belly and remove intestines and organs, until you're confident that it will not rot from inside." ) );
+                    break;
+                case 3:
+                    p->add_msg_if_player( m_good, _( "You remove guts and excess parts, preparing the corpse for later use." ) );
+                    break;
+            }
+            corpse_item.set_flag( "FIELD_DRESS" );
+
+            g->m.add_splatter( type_gib, p->pos(), rng( corpse->size + 2, ( corpse->size + 1 ) * 2 ) );
+            g->m.add_splatter( type_blood, p->pos(), rng( corpse->size + 2, ( corpse->size + 1 ) * 2 ) );
+            for( int i = 1; i <= corpse->size; i++ ) {
+                g->m.add_splatter_trail( type_gib, p->pos(), random_entry( g->m.points_in_radius( p->pos(), corpse->size + 1 ) ) );
+                g->m.add_splatter_trail( type_blood, p->pos(), random_entry( g->m.points_in_radius( p->pos(), corpse->size + 1 ) ) );
+            }
 
+        }
+        if( corpse->size > MS_TINY &&
+            query_yn( _( "Quarter %s to reduce it's weight and volume?  This will ruin it's skin." ), corpse_item.tname().c_str() ) ) {
```

Comment:
Correction: `its`
Suggestion: `Quarter corpse of %s to reduce the weight and volume?  This will ruin the hide or pelt.`